File: 프로그래머스/KaKao/택배 배달과 수거하기.swift
// 2023 KAKAO BLIND RECRUITMENT 택배 배달과 수거하기
import Foundation

func solution(_ cap:Int, _ n:Int, _ deliveries:[Int], _ pickups:[Int]) -> Int64 {
    var answer: Int64 = 0
    let deliveries = Array(deliveries.reversed())
    let pickups = Array(pickups.reversed())
    
    var currentDelivery = 0
    var currentPickup = 0
    
    for i in 0..<n {
        currentDelivery += deliveries[i]
        currentPickup += pickups[i]
        
        while currentDelivery > 0 || currentPickup > 0 {
            currentDelivery -= cap
            currentPickup -= cap

            answer += Int64((n - i) * 2)
        }
    }
    return answer
}

File: 프로그래머스/KaKao/택배 배달과 수거하기.py
# 2023 KAKAO BLIND RECRUITMENT 택배 배달과 수거하기
def solution(cap, n, deliveries, pickups):
    answer = 0
    deliveries = deliveries[::-1]
    pickups = pickups[::-1]

    curD, curP = 0, 0

    for i in range(n):
        curD += deliveries[i]
        curP += pickups[i]

        while curD > 0 or curP > 0:
            curD -= cap
            curP -= cap

            answer += 2*(n - i)
        # print(curD, curP)
    return answer
